scope.go

```diff
@@ -151,15 +152,11 @@ func (s *scope) incQueued() error {
 	defer queueSize.Dec()
 
 	// Try starting the request during the given duration.
-	d := s.maxQueueTime()
-	dSleep := d / 10
-	if dSleep > time.Second {
-		dSleep = time.Second
-	}
-	if dSleep < time.Millisecond {
-		dSleep = time.Millisecond
-	}
-	deadline := time.Now().Add(d)
+	sleep, deadline := s.calculateQueueDeadlineAndSleep()
+	return s.tryRunRequest(sleep, deadline, labels)
+}
+
+func (s *scope) tryRunRequest(sleep time.Duration, deadline time.Time, labels prometheus.Labels) error {
```

Comment:
IMHO, the name of the function is a bit misleading since we might think it runs the SQL or HTTP request.

A better name (but a bit longish) would be "wait until it's allowed to start"